File: test/fixtures/node-got/multiline_post_with_data.js
import got from 'got';

const response = await got('http://localhost:28139/echo/html/', {
  headers: {
    'Origin': 'http://fiddle.jshell.net'
  },
  form: {
    'msg1': 'value1',
    'msg2': 'value2'
  },
  allowGetBody: true
});

File: test/fixtures/kotlin-ktor/multiline_post_with_data.kt
import io.ktor.client.HttpClient
import io.ktor.client.request.forms.FormDataContent
import io.ktor.client.request.get
import io.ktor.client.request.header
import io.ktor.client.request.setBody
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient()

runBlocking {
    val response = client.get("http://localhost:28139/echo/html/") {
        setBody(
    FormDataContent(
        formData {
                append("msg1", "value1")
                append("msg2", "value2")
            }
    )
)
        header("Origin", "http://fiddle.jshell.net")
        header("Content-Type", "application/x-www-form-urlencoded")
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
